stdlib/public/core/UnsafeRawPointer.swift.gyb

```diff
@@ -0,0 +1,557 @@
+//===--- UnsafeRawPointer.swift.gyb --------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+%import gyb
+
+% for mutable in (True, False):
+%  Self = 'UnsafeMutableRawPointer' if mutable else 'UnsafeRawPointer'
+%  a_Self = 'an `UnsafeMutableRawPointer`' if mutable else 'an `UnsafeRawPointer`'
+%  Mutable = 'Mutable' if mutable else ''
+
+/// A raw pointer for accessing untyped data. This provides no
+/// automatic memory management, no type safety, and no alignment
+/// guarantees. This implements Strideable to provide a view
+/// of byte-addressable memory.
+@_fixed_layout
+public struct Unsafe${Mutable}RawPointer : Strideable, Hashable, _Pointer {
+  /// The underlying raw pointer.
+  public let _rawValue: Builtin.RawPointer
```

Comment:
I'm happy to make _rawValue internal. I have no idea what will break if we also make UnsafePointer._rawValue internal. There's no comment explaining why it was made public.